service/gcs/core/gcs/gcs.go

```diff
@@ -553,18 +562,28 @@ func (c *gcsCore) RegisterContainerExitHook(id string, exitHook func(oslayer.Pro
 // process may have multiple exit hooks registered for it.
 // This function works for both processes that are running in a container, and
 // ones that are running externally to a container.
-func (c *gcsCore) RegisterProcessExitHook(pid int, exitHook func(oslayer.ProcessExitState)) error {
-	c.processCacheMutex.Lock()
-	defer c.processCacheMutex.Unlock()
-	c.externalProcessCacheMutex.Lock()
-	defer c.externalProcessCacheMutex.Unlock()
+func (c *gcsCore) RegisterProcessExitHook(id string, pid int, exitHook func(oslayer.ProcessExitState)) error {
+	var (
+		entry *processCacheEntry
+		ok    bool
+	)
 
-	var entry *processCacheEntry
-	var ok bool
-	entry, ok = c.processCache[pid]
-	if !ok {
-		entry, ok = c.externalProcessCache[pid]
-		if !ok {
+	if id == "" {
```

Comment:
Do we actually expect the HCS to send an empty containerID string for external processes? I didn't know about this behavior, just making sure this is something we can definitely rely on.